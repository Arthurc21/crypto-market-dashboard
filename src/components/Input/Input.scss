.input {
	width: fit-content;

	& .label-container {
		display: flex;
		margin-bottom: 0.25rem;
		font-size: 12px;
		line-height: 14px;
		font-weight: 700;

		& .required-icon {
			color: #970c25;
		}
	}

	& .input-container {
		& input {
            background: #f2f2f2;
            border: 0;
            border-radius: 4px;
            box-sizing: border-box;
			width: 100%;
			color: #333333;
			font-weight: 400;
			font-size: 12px;
			letter-spacing: 0.43px;
            padding: 15px;
            outline: 0;
		}
	}
}
