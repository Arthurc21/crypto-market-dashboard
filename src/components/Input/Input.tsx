import React, { useState, useEffect, forwardRef } from 'react'
import './Input.scss'

interface InputProps {
	id: string
	label: string
	value: string
	required?: boolean
	placeholder?: string
	onChange(value: string): void
}

export const Input = forwardRef<HTMLInputElement, InputProps>(
	({ id, label, value, required = false, placeholder, onChange }, ref?) => {
		const elementId = `${id}-input`

		return (
			<div className="input">
				<div className="label-container">
					<label className="input-label" htmlFor={elementId}>
						{label}
					</label>
					{required && <div className="required-icon">&nbsp;*</div>}
				</div>
				<div className="input-container">
					<input
						type="text"
						value={value}
						placeholder={placeholder}
						onChange={(e) => onChange(e.target.value)}
					/>
				</div>
			</div>
		)
	}
)
