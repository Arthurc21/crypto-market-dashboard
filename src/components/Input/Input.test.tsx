import { create } from 'react-test-renderer'
import { Input } from './Input'

describe('Input Testing', () => {
	const onCheckFunc = jest.fn()
	test('input snapshot', () => {
		let input = create(
			<Input
				id={'inputTest'}
				label={'inputTest'}
				value={'inputTest'}
				required={true}
				placeholder={'test'}
				onChange={onCheckFunc}
			/>
		)
		expect(input.toJSON()).toMatchSnapshot()
	})
})
