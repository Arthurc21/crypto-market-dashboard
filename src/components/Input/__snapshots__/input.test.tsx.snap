// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Testing input snapshot 1`] = `
<div
  className="input label-position-top"
>
  <div
    className="label-container position-top"
  >
    <label
      className="input-label"
      htmlFor="inputTest-input"
    >
      inputTest
    </label>
    <div
      className="required-icon"
    >
      Â *
    </div>
  </div>
  <div
    className="input-container"
  >
    <div
      className="input-wrapper"
    >
      <input
        className=""
        onChange={[Function]}
        placeholder="test"
        type="text"
        value="inputTest"
      />
    </div>
    <div
      className="error-message-container"
    >
      <p
        className="error-message"
      />
    </div>
  </div>
</div>
`;
